zerver/views/webhooks/foursquare.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from typing import Dict, Any, Iterable, Optional
+
+BODY_TEMPLATE = ("Food nearby {displayString} coming right up\n\n"
+                 "{name1}\n{formattedAddress1}\n{text1}\n\n"
+                 "{name2}\n{formattedAddress2}\n{text2}\n\n"
+                 "{name3}\n{formattedAddress3}\n{text3}")
+
+@api_key_only_webhook_view('FourSquare')
+@has_request_variables
+def api_foursquare_webhook(request, user_profile, client,
+                           payload=REQ(argument_type='body'),
+                           stream=REQ(default='test'),
+                           topic=REQ(default='FourSquare')):
+                                # type: (HttpRequest, UserProfile, Client, Dict[str, Iterable[Dict[str, Any]]]) -> HttpResponse
+    # try to add the Foursquare results
+    try:
+        # JSON list of venues
+        venue_list = payload['response']['groups'][0]['items']
+        # seperation into 3 venues and defines location
+        place1 = venue_list[0]['venue']
```

Comment:
could we create  functions that returns location and full address based on name? 
It will simplify our code